### DISCUSSION TITLE
v1.84.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#4777] [Announcements]
# 1.84.0

Welcome to release `v1.84.0`. This release brings many quality-of-life improvements, a new mobile app bar, and bug fixes for machine learning. I hope you enjoy this release.

## Highlights

- New mobile app bar and user profile screen
- Shared link with password option
- Shared album activity (web)
- Custom scan interval for external library
- Notable fix: Cannot download machine learning models from the S3 bucket

### New mobile app bar and user profile screen

Thanks to @shenlong-tanwen, we now have the new aesthetically pleasing app bar and user profile panel. The trash button can now be found on the `library` page.

![Mobile App Bar](https://github.com/immich-app/immich/assets/27055614/dd6226f3-94c6-4a1d-86c3-5f298584eeef)

### Shared link with password option

We added an optional `password` option for the shared link. Thanks, @jarvis2f, for your work!

![Shared link password](https://github.com/immich-app/immich/assets/27055614/85b70b07-de56-4942-8ba0-fb13a76f4fda)

### Shared album activity

With the fantastic and tight collaboration between @martabal and @jrasm91, after two days of work, you can now comment on a photo or video in the shared album, and the shared users can all read the feed. This feature is web-only at the current stage.

![Activity](https://github.com/immich-app/immich/assets/27055614/40a9d105-1738-4b51-b344-c8bb9525e12c)

### Custom scan interval for external library

You can now define a custom interval for the trigger or external library rescan. Thank you, @danieldietzler, for the enhancement to the feature.

![image](https://github.com/immich-app/immich/assets/27055614/91fc1764-a305-4afb-8f74-017d960f4b27)

- - - -

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 1FvEp6P6NM8EZEkpGUFAN2LqJ1gxusNxZX

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### Web
* feat(web): added toggle for Sharing button in the sidebar by @upsetdog in https://github.com/immich-app/immich/pull/4674
* feat(web,server): activity by @martabal in https://github.com/immich-app/immich/pull/4682
* fix(web): table headers when there's no album by @martabal in https://github.com/immich-app/immich/pull/4673
* fix(web): scrollbar does not show all years by @alextran1502 in https://github.com/immich-app/immich/pull/4684
* fix(web): Improve UI/UX for shared link form by @alextran1502 in https://github.com/immich-app/immich/pull/4685
* fix(web): sidebar setting not updating when there is a new property added to the data payload by @alextran1502 in https://github.com/immich-app/immich/pull/4708
* fix(web): multiple improvements for people page (1) by @martabal in https://github.com/immich-app/immich/pull/4717
* fix(web): no preload repair report by @jrasm91 in https://github.com/immich-app/immich/pull/4749

### Mobile
* feat(mobile): new app bar by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4687
* fix(mobile): handle shared assets actions in the viewer by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4679
* fix(mobile): first char miss in new description by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4697
* fix(mobile): cache key for assets from DTO by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4699
* fix(mobile): don't show lens info if it's not available by @waclaw66 in https://github.com/immich-app/immich/pull/4737
* fix(mobile): handle non-standard exposure time string by @shenlong-tanwen in https://github.com/immich-app/immich/pull/4589

### Server
* feat(server): Add an optional password option for shared links. by @jarvis2f in https://github.com/immich-app/immich/pull/4655
* feat(server): hardware video acceleration for Rockchip SOCs via RKMPP by @fyfrey in https://github.com/immich-app/immich/pull/4645
* feat(server): custom library scanning interval by @danieldietzler in https://github.com/immich-app/immich/pull/4390
* fix(server): Handle errors from OAuth Discovery by @samip5 in https://github.com/immich-app/immich/pull/4678
* fix(server): do not leak people by @martabal in https://github.com/immich-app/immich/pull/4710
* fix(server): better fix for the OAuth Discovery errors by @samip5 in https://github.com/immich-app/immich/pull/4695
* fix(server): Correctly set album start and end dates by @adamantike in https://github.com/immich-app/immich/pull/4698
* fix(server): import scheduler module by @alextran1502 in https://github.com/immich-app/immich/pull/4766
* refactor(server): auth delete device by @jrasm91 in https://github.com/immich-app/immich/pull/4720
* refactor(server): user core by @jrasm91 in https://github.com/immich-app/immich/pull/4722
* refactor(server): user profile picture by @jrasm91 in https://github.com/immich-app/immich/pull/4728
* chore(server): remove user count endpoint by @jrasm91 in https://github.com/immich-app/immich/pull/4724
* refactor(server): user core by @jrasm91 in https://github.com/immich-app/immich/pull/4733
* fix(cli): ignore web socket when unavailable and skip metadata init by @jrasm91 in https://github.com/immich-app/immich/pull/4748

### Machine Learning
* feat(ml): export clip models to ONNX and host models on Hugging Face by @mertalev in https://github.com/immich-app/immich/pull/4700

### Others
* docs: update milestone page by @martabal in https://github.com/immich-app/immich/pull/4683
* chore: add contribution section to readme by @Kalyanimhala in https://github.com/immich-app/immich/pull/4690
* docs: Fix typos by @Mayureshd-18 in https://github.com/immich-app/immich/pull/4713

## New Contributors
* @jarvis2f made their first contribution in https://github.com/immich-app/immich/pull/4655
* @upsetdog made their first contribution in https://github.com/immich-app/immich/pull/4674
* @Kalyanimhala made their first contribution in https://github.com/immich-app/immich/pull/4690
* @Mayureshd-18 made their first contribution in https://github.com/immich-app/immich/pull/4713

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.83.0...v1.84.0

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.84.0'>v1.84.0</a>.</em>

**Comment by ayykamp** [comment]
How do I comment? Is it something I have to explicitly enable?

**Reply by alextran1502** [reply]
No, you can do that from a shared album

**Reply by ayykamp** [reply]
The button should be in the bottom right, correct? Its not showing up for me (Have tried different browser and clearing cache)
![image](https://github.com/immich-app/immich/assets/32194363/b57f5c4b-fa12-425f-a2c5-0e846c5c4226)
![image](https://github.com/immich-app/immich/assets/32194363/afd6f4b9-f8b0-487a-ac33-0318f94ddd20)

**Reply by martabal** [reply]
Have you shared this album with other users ?

**Reply by ayykamp** [reply]
Oh sorry I thought this worked for albums shared via link too. My bad.
Thank you for your work :))

**Comment by Uberpainer** [comment]
As always a beautiful peace of work :)
Does it require me to log out and in again in the app to experience the "New mobile app bar and user profile screen"? No updates at least found in app store for iPhone.

**Reply by jrasm91** [reply]
I think when the release happens on GitHub, Alex submits the apps to the respective app stores, but when those get approved and pushed out can vary from a few minutes to a few days.
